src/object.rs

```diff
@@ -414,4 +419,60 @@ impl<T: IsA<Object>> ObjectExt for T {
             (Some(ptype), Some(type_)) => ptype == type_,
         }
     }
+
+    fn connect<'a, N, F>(&self, signal_name: N, after: bool, callback: F) -> Result<u64, BoolError>
+        where N: Into<&'a str>, F: Fn(&[Value]) -> Option<Value> + Send + Sync + 'static {
+        let signal_name: &str = signal_name.into();
+
+        unsafe {
+            let obj = self.to_glib_none().0;
+            let klass = (*obj).g_type_instance.g_class as *mut gobject_ffi::GTypeClass;
+            let type_ = (*klass).g_type;
```

Comment:
Because this gives you the static type of whatever was passed in here. So if you have a GtkButton and cast it to a GtkWidget, and pass it in here then, you would get the type of GtkWidget (which does not have e.g. the "clicked" signal).

Or more problematic, in case of GStreamer the types of objects from plugins don't even have bindings and you always pass them around as GstElements.